### PR TITLE
Ticket 25999

### PR DESCRIPTION
**Pull request by dmifedorenko**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #25999
| License       | MIT

### CODE REVIEW

File: CHANGELOG-3.4.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ in 3.4 minor versions.
 To get the diff for a specific change, go to https://github.com/symfony/symfony/commit/XXX where XXX is the change hash
 To get the diff between two versions, go to https://github.com/symfony/symfony/compare/v3.4.0...v3.4.1
 
+* 3.4.x (2018-xx-xx)
+
+ * bug #25999 [DependencyInjection] Huge performance degradation of ReflectionClassResource and FileResource on big projects inside Docker For Mac
+
```
Comment (iltar) [inline CHANGELOG-3.4.md]: This file is updated automatically when a new symfony version is released. Can you please revert this change?

File: src/Symfony/Component/Config/Resource/FileResource.php [hunk 2]
```diff
@@ -60,7 +60,9 @@ public function getResource()
      */
     public function isFresh($timestamp)
     {
-        return file_exists($this->resource) && @filemtime($this->resource) <= $timestamp;
+        $filemtime = @filemtime($this->resource);
+
+        return $filemtime && $filemtime <= $timestamp;
```
Comment (iltar) [inline src/Symfony/Component/Config/Resource/FileResource.php]: These changes can probably be fixed in a lower branch.

File: src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php [hunk 3]
```diff
@@ -34,12 +34,13 @@ public function __construct($class, $path, array $autowiringMetadata)
 
     public function isFresh($timestamp)
     {
-        if (!file_exists($this->filePath)) {
+        $filemtime = @filemtime($this->filePath);
+        if (!$filemtime) {
```
Comment (iltar) [inline src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php]: The changes in this file can probably also be applied on a lower branch.

**Review by linaori — CHANGES_REQUESTED** [review]
To fix the changes in a lower branch, you'll have to make different pull requests. Find the lowest maintained branch (2.7 in this case) that supports your patch for those lines and base the pull request on that branch. You mentioned that the `ReflectionClassResource` does not exist in 2.7 yet, so those changes need to be applied in a higher branch than 2.7 (meaning multiple pull requests).

File: src/Symfony/Component/Config/Resource/FileResource.php (hunk 2 above)
Comment (sroze) [reply src/Symfony/Component/Config/Resource/FileResource.php]: You can probably inline the definition as well, i.e. 
```php
return false !== ($filemtime = @filemtime($this->resource))&& $filemtime <= $timestamp;
```

File: src/Symfony/Component/Config/Resource/ReflectionClassResource.php [hunk 4]
```diff
@@ -37,11 +37,12 @@ public function isFresh($timestamp)
         }
 
         foreach ($this->files as $file => $v) {
-            if (!file_exists($file)) {
+            $filemtime = @filemtime($file);
+            if (!$filemtime) {
```
Comment (sroze) [inline src/Symfony/Component/Config/Resource/ReflectionClassResource.php]: inline?

```php
if (false === ($filemtime = @filemtime($file))) {
```

**Comment by sroze** [comment]
Also, it would be good to give the pull-request a nice title, such as "Only using `filemtime` to check file freshness" or something like that; this will help everybody to know roughly what this pull-request is about.

File: src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php (hunk 3 above)
Comment (Simperfit) [reply src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php]: same as @sroze you can inline all this condition

**Comment by Simperfit** [comment]
Travis failure seems unrelated to the changes
